dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/support/ForkingClusterInvoker.java

```diff
@@ -96,7 +93,7 @@ public Result doInvoke(final Invocation invocation, List<Invoker<T>> invokers, L
                 });
             }
             try {
-                Object ret = ref.poll(timeout, TimeUnit.MILLISECONDS);
+                Object ret = ref.poll(Integer.MAX_VALUE, TimeUnit.MILLISECONDS);
```

Comment:
What is the purpose change this line? Will this cause NPE?

Reply:
Yes，if poll method use the same timeout with the provider，when provider timeout the poll method used in forkingClusterInvoker will time out earlier than the provider side, so it will get null from the queue and throw NPE in the subsequent process.